prow/cmd/status-reconciler/main.go

```diff
@@ -130,8 +139,17 @@ func main() {
 		logrus.WithError(err).Fatal("Error getting kube client.")
 	}
 
-	c := statusreconciler.NewController(o.continueOnError, sets.NewString(o.addedPresubmitBlacklist.Strings()...), prowJobClient, githubClient, configAgent, pluginAgent)
+	opener, err := io.NewOpener(context.Background(), o.gcsCredentialsFile)
```

Comment:
Could we add a timeout to each call so that we don't deadlock? ref #13816 

This opener needs to be closed

Reply:
Thank you for the review on the deadlock possibility. Done.

Regarding the opener, both `Load` and `Save` operations close the files they open. The `opener` itself uses is a GCS client which says does not need to be closed on program exit. Do you recommend updating the `pkg/io/opener.go` itself to add a `Close` method?

Reply:
Ah no, if it's not intended to be closed then that's fine -- bad assumption on my end based on the name